src/dune_lang/action.ml

```diff
@@ -349,6 +394,12 @@ let rec encode =
   | Cram script -> List [ atom "cram"; sw script ]
   | Patch i -> List [ atom "patch"; sw i ]
   | Substitute (i, o) -> List [ atom "substitute"; sw i; sw o ]
+  | Withenv (ops, t) ->
+    List
+      [ atom "withenv"
```

Comment:
`with-env` would probably fit better with e.g. `write-file`.

Reply:
I would prefer the `-` version. We can create an issue to rename `setenv` to `set-env`.

Reply:
I was looking through the docs and we mostly use `_` (which is not really typical in Lisp, but its better to be consistent than typical I'd say).